src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3377,9 +3427,7 @@ private bool HasTopLevelNullabilityConversion(TypeSymbolWithAnnotations source,
                 case ConversionKind.Boxing:
                     if (!operandType.IsNull && operandType.IsValueType)
                     {
-                        // https://github.com/dotnet/roslyn/issues/29959 Should we worry about a pathological case of boxing nullable value known to be not null?
-                        //       For example, new int?(0)
-                        resultAnnotation = operandType.IsNullableType() ? NullableAnnotation.Nullable : NullableAnnotation.NotNullable;
```

Comment:
>operandType.IsNullableType() [](start = 43, length = 28)

I think we should keep the ```IsNullableType``` check. Malformed annotations in metadata can have nullable annotation of a non-nullable value type.